app/models/hmcts_common_platform/defendant.rb

```diff
@@ -77,5 +104,16 @@ def to_builder
     def person_defendant
       HmctsCommonPlatform::PersonDefendant.new(data[:personDefendant])
     end
+
+    def valid_maat_reference?
+      unique_maat_reference.present? && !unique_maat_reference.start_with?("Z")
+    end
+
+    def unique_maat_reference
+      refs = offences.map(&:laa_application_reference).uniq.compact
+      raise "Too many maat references" if refs.size > 1
+
+      refs&.first
+    end
```

Comment:
Could these be refactored? Maybe combined? It looks like `unique_maat_reference` will get called 3 times for each call of `maat_reference`